packages/next/next-server/lib/router/router.ts

```diff
@@ -903,11 +909,11 @@ export default class Router implements BaseRouter {
     })
   }
 
-  abortComponentLoad(as: string): void {
+  abortComponentLoad(): void {
     if (this.clc) {
       const e = new Error('Route Cancelled')
       ;(e as any).cancelled = true
-      Router.events.emit('routeChangeError', e, as)
+      Router.events.emit('routeChangeError', e, this._lastStarted)
```

Comment:
I'm not particularly fond of this shared state, I preferred when it was passed into the function. I think we can fixup this logic.

Reply:
even when passed, it needs the value from the previous call to the function that passes it

Reply:
I made it pass the url to the function again. I also renamed the state to `_inFlightRoute` to make the intent more clear.